### ISSUE TITLE
Transformation throws error instead of diagnostic for unsupported node

### ISSUE THREAD

**Issue by Perryvw** [TypeScriptToLua/TypeScriptToLua#860]
I was playing around on the playground and discovered an unexpected error while trying to transform the following code:

```ts
const a = () => {
    foo: "bar",
    baz: "buz"
};
const { foo, baz } = a;
```
[Playground link](https://typescripttolua.github.io/play/#code/MYewdgzgLgBAhjAvDAFASiQPhgbwFAyEwBmIIAXDAEQBGcATlQDQFF0BeltAru1XgF8A3HjyhIsHCTJMYHGAKTwhQA)

Expected behavior:
No uncaught exception. Instead expected a unsupported node diagnostic.

**Comment by lolleko** [comment]
Related: #844 #691

**Comment by Perryvw** [comment]
Fixed with #861
